### PR TITLE
op-guide, media: add TiKV metrics

### PR DESCRIPTION
**Pull request by rleungx**
This PR adds the TiKV metrics on TiKV dashboard.

Related PR: pingcap/docs-cn#1015

### CODE REVIEW

File: op-guide/dashboard-tikv-info.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+To understand the key metrics displayed on the Overview dashboard, check the following table:
+
+Row name | Panel name | Description | Normal range
+---------------- | ---------------- | ---------------------------------- | --------------
+Cluster | Store size | The storage size of each TiKV instance |
+Cluster | Available size | The available capacity size of each TiKV instance |
+Cluster | Capacity size | The capacity size of each TiKV instance |
+Cluster | CPU | The cpu usage of each TiKV instance |
+Cluster | Memory | The memory usage of each TiKV instance |
+Cluster | IO utilization | The io utilization of each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Cluster | IO utilization | The I/O utilization of each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+Scheduler | Scheduler priority commands | The count of different priority commands |
+Scheduler | Scheduler pending commands | The count of pending commands on each TiKV instance |
+Scheduler - batch_get | Scheduler stage total | The total number of commands on each stage in batch_get command | There should not be lots of errors in a short time.
+Scheduler - batch_get | Scheduler command duration | The time consumed when execting batch_get command | It should be less than `1s`.
+Scheduler - batch_get | Scheduler latch wait duration | The time which is caused by latch wait in batch_get command | It should be less than `1s`.
+Scheduler - batch_get | Scheduler keys read | The count of keys read by a batch_get command |
+Scheduler - batch_get | Scheduler keys written | The count of keys written by a batch_get command |
+Scheduler - batch_get | Scheduler scan details | The keys scan details of each CF when executing batch_get command |
+Scheduler - batch_get | Scheduelr scan details [lock] | The keys scan details of lock CF when executing batch_get command |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Scheduler - batch_get | Scheduler scan details [lock] | The keys scan details of lock CF when executing batch_get command |
```

File: op-guide/dashboard-tikv-info.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+Scheduler - batch_get | Scheduelr scan details [write] | The keys scan details of write CF when executing batch_get command |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Scheduler - batch_get | Scheduler scan details [write] | The keys scan details of write CF when executing batch_get command |
```

File: op-guide/dashboard-tikv-info.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+Scheduler - batch_get | Scheduelr scan details [default] | The keys scan details of default CF when executing batch_get command |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Scheduler - batch_get | Scheduler scan details [default] | The keys scan details of default CF when executing batch_get command |
```

File: op-guide/dashboard-tikv-info.md [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+Scheduler - cleanup | Scheduler stage total | The total number of commands on each stage in cleanup command | There should not be lots of errors in a short time. 
+Scheduler - cleanup | Scheduler command duration | The time consumed when execting cleanup command | It should be less than `1s`.
+Scheduler - cleanup | Scheduler latch wait duration | The time which is caused by latch wait in cleanup command | It should be less than `1s`.
+Scheduler - cleanup | Scheduler keys read | The count of keys read by a cleanup command |
+Scheduler - cleanup | Scheduler keys written | The count of keys written by a cleanup command |
+Scheduler - cleanup | Scheduler scan details | The keys scan details of each CF when executing cleanup command |
+Scheduler - cleanup | Scheduelr scan details [lock] | The keys scan details of lock CF when executing cleanup command |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Scheduler - cleanup | Scheduler scan details [lock] | The keys scan details of lock CF when executing cleanup command |
```

File: op-guide/dashboard-tikv-info.md [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+## Key metrics description
+
+To understand the key metrics displayed on the Overview dashboard, check the following table:
+
+Row name | Panel name | Description | Normal range
+---------------- | ---------------- | ---------------------------------- | --------------
+Cluster | Store size | The storage size of each TiKV instance |
+Cluster | Available size | The available capacity size of each TiKV instance |
+Cluster | Capacity size | The capacity size of each TiKV instance |
+Cluster | CPU | The cpu usage of each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Cluster | CPU | The CPU usage of each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+Cluster | Memory | The memory usage of each TiKV instance |
+Cluster | IO utilization | The io utilization of each TiKV instance |
+Cluster | MBps | The total bytes and keys of read and write on each TiKV instance |
+Cluster | QPS | The QPS of different kinds of commands in each TiKV instance |
+Cluster | Errps | The total number of the gRPC message failure |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Cluster | Errps | The total number of the gRPC message failures |
```

File: op-guide/dashboard-tikv-info.md [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+Server | Region | The number of Regions on each TiKV instance |
+Server | CF size | The total size of each column family |
+Server | Store size | The storage size of each TiKV instance |
+Server | Channel full | The total number of channel full errors on each TiKV instance | It should be `0` in normal case.
+Server | Server report failures | The total number of reporting failure messages |
+Server | Region average written keys | The average rate of writing keys to Regions on each TiKV instance |
+Server | Region average written bytes | The average rate of writing bytes to Regions on each TiKV instance |
+Server | Active written leaders | The number of active leaders on each TiKV instance |
+Server | Approximate Region size | The approximate Region size | 
+Raft IO | Apply log duration | The time consumed when Raft apply log |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft IO | Apply log duration | The time consumed when Raft applies log |
```

File: op-guide/dashboard-tikv-info.md [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+Raft IO | Apply log duration per server | The time consumed when Raft apply log on each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft IO | Apply log duration per server | The time consumed when Raft applies log on each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+Raft IO | Append log duration | The time consumed when Raft append log |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft IO | Append log duration | The time consumed when Raft appends log |
```

File: op-guide/dashboard-tikv-info.md [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+Raft IO | Append log duration per server | The time consumed when Raft append log on each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft IO | Append log duration per server | The time consumed when Raft appends log on each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+Raft process | Ready handled | The count of different ready type of Raft |
+Raft process | Process ready duration per server | The time consumed when peer processing ready in Raft | It should be less than `2s` in `.9999`.
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft process | Process ready duration per server | The time consumed when the peer processes ready in Raft | It should be less than `2s` in `.9999`.
```

File: op-guide/dashboard-tikv-info.md [hunk 13]
```diff
@@ -0,0 +1,315 @@
...
+Raft process | Process tick duration per server | The time consumed when peer processing tick in Raft |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft process | Process tick duration per server | The time consumed when the peer processes tick in Raft |
```

File: op-guide/dashboard-tikv-info.md [hunk 14]
```diff
@@ -0,0 +1,315 @@
...
+Raft process | 0.99 Duration of raftstore events | The time consumed of raftstore events in `.99` |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft process | 0.99 Duration of raftstore events | The time consumed by raftstore events in `.99` |
```

File: op-guide/dashboard-tikv-info.md [hunk 15]
```diff
@@ -0,0 +1,315 @@
...
+Raft message | Sent messages per server | The number of sending Raft messages of each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft message | Sent messages per server | The number of Raft messages sent by each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 16]
```diff
@@ -0,0 +1,315 @@
...
+Raft message | Flush messages per server | The number of flushing Raft messages of each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft message | Flush messages per server | The number of Raft messages flushed by each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 17]
```diff
@@ -0,0 +1,315 @@
...
+Raft message | Receive messages per server | The number of receiving Raft messages of each TiKV instance |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft message | Receive messages per server | The number of Raft messages received by each TiKV instance |
```

File: op-guide/dashboard-tikv-info.md [hunk 18]
```diff
@@ -0,0 +1,315 @@
...
+Raft message | Messages | The number of sending different types of Raft messages  |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft message | Messages | The number of different types of Raft messages that are sent |
```

File: op-guide/dashboard-tikv-info.md [hunk 19]
```diff
@@ -0,0 +1,315 @@
...
+Raft message | Vote | The total number of vote messages is sent in Raft |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft message | Vote | The total number of vote messages that are sent in Raft |
```

File: op-guide/dashboard-tikv-info.md [hunk 20]
```diff
@@ -0,0 +1,315 @@
...
+Raft message | Raft dropped messages | The number of dropping different types of Raft messages |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft message | Raft dropped messages | The number of different types of Raft messages that are dropped |
```

File: op-guide/dashboard-tikv-info.md [hunk 21]
```diff
@@ -0,0 +1,315 @@
...
+Raft propose | Raft proposals per ready | The proposal count of all Regions in a mio tick |
+Raft propose | Raft read/write proposals | The total number of different kinds of proposals |
+Raft propose | Raft read proposals per server | The number of read proposals which are made by each TiKV instance |
+Raft propose | Raft write proposals per server | The number of write proposals which are made by each TiKV instance |
+Raft propose | Propose wait duration | The wait time of each proposal |
+Raft propose | Propose wait duration per server | The wait time of each proposal in each TiKV instance |
+Raft propose | Raft log speed | The speed of peers propose log |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Raft propose | Raft log speed | The speed that peers propose log |
```

File: op-guide/dashboard-tikv-info.md [hunk 22]
```diff
@@ -0,0 +1,315 @@
...
+Local reader | Local read requests batch size | The batch size of local read requests |
+Storage | Storage command total | The total count of different kinds of commands received |
+Storage | Storage async request error | The total number of engine asynchronous request errors |
+Storage | Storage async snapshot duration | The time consumed by processing asynchronous snapshot requests | It should be less than `1s` in `.99`.
+Storage | Storage async write duration | The time consumed by processing asynchronous write requests | It should be less than `1s` in `.99`.
+Scheduler | Scheduler stage total | The total number of commands on each stage | There should not be lots of errors in a short time.
+Scheduler | Scheduler priority commands | The count of different priority commands |
+Scheduler | Scheduler pending commands | The count of pending commands on each TiKV instance |
+Scheduler - batch_get | Scheduler stage total | The total number of commands on each stage in batch_get command | There should not be lots of errors in a short time.
+Scheduler - batch_get | Scheduler command duration | The time consumed when execting batch_get command | It should be less than `1s`.
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
Scheduler - batch_get | Scheduler command duration | The time consumed when executing batch_get command | It should be less than `1s`.
```

File: op-guide/dashboard-tikv-info.md [hunk 23]
```diff
@@ -0,0 +1,315 @@
...
+RocksDB - raft | Stall conditions changed of each CF | Stall conditions changed of each column family |
+gRPC | gRPC message count | The count of different kinds of gRPC message |
+gRPC | gRPC message failed | The count of different kinds of gRPC message which is failed |
+gRPC | 99% gRPC message duration | The execution time of gRPC message |
+gRPC | gRPC GC message count | The count of gRPC GC message |
+gRPC | 99% gRPC KV GC message duration | The execution time of gRPC GC message |
+PD | PD requests | The count of requests that TiKV sends to PD |
+PD | PD request duration (average) | The time consumed by requests that TiKV sends to PD |
+PD | PD heartbeats | The total number of PD heartbeat messages |
+PD | PD validate peers | The total number of PD worker validate peer task |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
PD | PD validate peers | The total number of peers validated by the PD worker |
```

File: op-guide/dashboard-tikv-info.md [hunk 24]
```diff
@@ -0,0 +1,315 @@
...
+RocksDB - kv | Ingest SST duration seconds | The time consumed when ingesting SST files |
+RocksDB - kv | Stall conditions changed of each CF | Stall conditions changed of each column family |
+RocksDB - raft | Get operations | The count of get operations |
+RocksDB - raft | Get duration | The time consumed when execting get operation |
+RocksDB - raft | Seek operations | The count of seek operations |
+RocksDB - raft | Seek duration | The time consumed when execting seek operation |
+RocksDB - raft | Write operations | The count of write operations |
+RocksDB - raft | Write duration | The time consumed when execting write operation |
+RocksDB - raft | WAL sync operations | The count of WAL sync operations |
+RocksDB - raft | WAL sync duration | The time consumed when execting WAL sync operation |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
RocksDB - raft | WAL sync duration | The time consumed when executing WAL sync operation |
```

File: op-guide/dashboard-tikv-info.md [hunk 25]
```diff
@@ -0,0 +1,315 @@
...
+RocksDB - kv | Compaction pending bytes | The pending bytes when executing compaction |
+RocksDB - kv | Read amplification | The read amplification in each TiKV instance |
+RocksDB - kv | Compression ratio | The compression ratio of each level |
+RocksDB - kv | Number of snapshots | The number of snapshot of each TiKV instance |
+RocksDB - kv | Oldest snapshots duration | The time that the oldest unreleased snapshot survivals |
+RocksDB - kv | Number files at each level | The number of SST files for different column families in each level |
+RocksDB - kv | Ingest SST duration seconds | The time consumed when ingesting SST files |
+RocksDB - kv | Stall conditions changed of each CF | Stall conditions changed of each column family |
+RocksDB - raft | Get operations | The count of get operations |
+RocksDB - raft | Get duration | The time consumed when execting get operation |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
RocksDB - raft | Get duration | The time consumed when executing get operation |
```

File: op-guide/dashboard-tikv-info.md [hunk 26]
```diff
@@ -0,0 +1,315 @@
...
+RocksDB - raft | Seek operations | The count of seek operations |
+RocksDB - raft | Seek duration | The time consumed when execting seek operation |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
RocksDB - raft | Seek duration | The time consumed when executing seek operation |
```

File: op-guide/dashboard-tikv-info.md [hunk 27]
```diff
@@ -0,0 +1,315 @@
...
+RocksDB - raft | Write operations | The count of write operations |
+RocksDB - raft | Write duration | The time consumed when execting write operation |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
RocksDB - raft | Write duration | The time consumed when executing write operation |
```

File: op-guide/dashboard-tikv-info.md [hunk 28]
```diff
@@ -0,0 +1,315 @@
...
+RocksDB - raft | WAL sync operations | The count of WAL sync operations |
+RocksDB - raft | WAL sync duration | The time consumed when execting WAL sync operation |
+RocksDB - raft | Compaction operations | The count of compaction and flush operations |
+RocksDB - raft | Compaction duration | The time consumed when execting compaction and flush operation |
```
Comment (lilin90) [inline op-guide/dashboard-tikv-info.md]:
```suggestion
RocksDB - raft | Compaction duration | The time consumed when executing compaction and flush operation |
```

**Comment by rleungx** [comment]
Done. PTAL @lilin90 @nolouch

**Comment by lilin90** [comment]
Please also check the comments about wording/description, and update them where necessary.

**Comment by lilin90** [comment]
I'll add this doc to TOC soon.

**Pull request merged**